From f697609fc6cb6a28d6b88498870da4ecdd62fd76 Mon Sep 17 00:00:00 2001
From: Shiqing Gao <shiqing.gao@intel.com>
Date: Thu, 8 Feb 2018 13:32:48 +0800
Subject: [PATCH 1112/2345] sos: add a config for VHM

seperate the config for CWP and VHM

SOS has to enable both CONFIG_CWP and CONFIG_CWP_VHM.
UOS only needs to enable CONFIG_CWP. VHM is not used in UOS.

Change-Id: I8529771e1943c18d790230533f7a4bcc84966350
Signed-off-by: Shiqing Gao <shiqing.gao@intel.com>
---
 arch/x86/cwp/Kconfig  |  4 ----
 arch/x86/cwp/cwp.c    |  3 +++
 drivers/Kconfig       |  2 ++
 drivers/Makefile      |  2 +-
 drivers/char/Makefile |  2 +-
 drivers/vhm/Kconfig   | 18 ++++++++++++++++++
 6 files changed, 25 insertions(+), 6 deletions(-)
 create mode 100644 drivers/vhm/Kconfig

diff --git a/arch/x86/cwp/Kconfig b/arch/x86/cwp/Kconfig
index e83e368..2880c87 100644
--- a/arch/x86/cwp/Kconfig
+++ b/arch/x86/cwp/Kconfig
@@ -7,10 +7,6 @@ config CWP
 	bool "Enable services run on CWP hypervisor"
 	depends on X86
 	depends on PARAVIRT
-	depends on DMA_CMA
-	depends on PCI_MSI
-	depends on !INTEL_IOMMU
-	depends on !VMAP_STACK
 	help
 	  This option is needed if were to run CWP services linux on top of
 	  CWP hypervisor.
diff --git a/arch/x86/cwp/cwp.c b/arch/x86/cwp/cwp.c
index b13ca87..09a6a1a 100644
--- a/arch/x86/cwp/cwp.c
+++ b/arch/x86/cwp/cwp.c
@@ -55,7 +55,10 @@ static uint32_t __init cwp_detect(void)
 static void __init cwp_init_platform(void)
 {
 	pv_cpu_ops.cpu_khz = cpu_khz_from_cwp;
+
+#ifdef CONFIG_CWP_VHM
 	pv_irq_ops.write_msi = cwp_write_msi_msg;
+#endif /* CONFIG_CWP_VHM */
 }
 
 static void cwp_pin_vcpu(int cpu)
diff --git a/drivers/Kconfig b/drivers/Kconfig
index 20d4e63..430f6b0 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -218,4 +218,6 @@ source "drivers/opp/Kconfig"
 source "drivers/vbs/Kconfig"
 
 source "drivers/cwp/Kconfig"
+
+source "drivers/vhm/Kconfig"
 endmenu
diff --git a/drivers/Makefile b/drivers/Makefile
index bd9ed2f..5856632 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -187,5 +187,5 @@ obj-$(CONFIG_FSI)		+= fsi/
 obj-$(CONFIG_TEE)		+= tee/
 obj-$(CONFIG_MULTIPLEXER)	+= mux/
 obj-$(CONFIG_SDW)		+= sdw/
-obj-$(CONFIG_CWP) 		+= vhm/
+obj-$(CONFIG_CWP_VHM) 		+= vhm/
 obj-$(CONFIG_CWP)		+= cwp/
diff --git a/drivers/char/Makefile b/drivers/char/Makefile
index 8c1247a..3c4b742 100644
--- a/drivers/char/Makefile
+++ b/drivers/char/Makefile
@@ -61,4 +61,4 @@ obj-$(CONFIG_TILE_SROM)		+= tile-srom.o
 obj-$(CONFIG_XILLYBUS)		+= xillybus/
 obj-$(CONFIG_POWERNV_OP_PANEL)	+= powernv-op-panel.o
 obj-$(CONFIG_RPMB)		+= rpmb/
-obj-$(CONFIG_CWP)		+= vhm/
+obj-$(CONFIG_CWP_VHM)		+= vhm/
diff --git a/drivers/vhm/Kconfig b/drivers/vhm/Kconfig
new file mode 100644
index 00000000..2ac3032
--- /dev/null
+++ b/drivers/vhm/Kconfig
@@ -0,0 +1,18 @@
+config CWP_VHM
+	bool "Intel CWP Hypervisor Virtio and Hypervisor service Module (VHM)"
+	depends on CWP
+	depends on DMA_CMA
+	depends on PCI_MSI
+	depends on !INTEL_IOMMU
+	depends on !VMAP_STACK
+	default n
+	---help---
+		This is the Virtio and Hypervisor service Module (VHM) for
+		Intel CWP hypervisor.
+
+		It is required for Service OS.
+		User OS doesn't need to have this config.
+
+		Say Y for SOS and say N for UOS.
+
+		If unsure, say N.
-- 
2.7.4

